roles/maas_nameserver/README.rst

```diff
@@ -0,0 +1,102 @@
+---
+
+# README for `maas_nameserver` Ansible Role
+
+## Overview
+The `maas_nameserver` role configures DNS domains and records in MAAS (Metal as a Service) based on an Ansible inventory. It manages DNS entries for Ceph nodes and their IPMI interfaces, ensuring only desired records and domains exist while cleaning up unwanted ones. This role depends on a `secrets` role to load encrypted MAAS API credentials using Ansible Vault.
+
+## Requirements
+- **Ansible**: Version 2.9 or higher
+- **MAAS CLI**: Installed on the target maas server
+- **Secrets Role**: A companion `secrets` role to load encrypted credentials
+
+## Role Structure
+```
+roles/
+  maas_nameserver/
+    defaults/
+      main.yml
+    meta/
+      main.yml
+    tasks/
+      main.yml
+    vars/
+      main.yml
+    README.md
+```
+
+## Dependencies
+- **`secrets` Role**: Defined in `meta/main.yml`, this role loads encrypted MAAS API credentials from a configurable path. Ensure the `secrets` role is available in your `roles/` directory.
+
+## Usage
+1. **Place the Role**: Ensure the `maas_namesever` and `secrets` roles are in your `roles/` directory.
+2. **Prepare Inventory**: Define your `maas` and `ceph` groups (e.g., in `/etc/ansible/hosts/tucson`):
+   ```ini
+   [ceph]
+   ceph001.internal.ceph.ibm.com ip=10.18.131.1 ipmi=10.18.139.1
+   ceph002.internal.ceph.ibm.com ip=10.18.131.2 ipmi=10.18.139.2
+   ceph003.internal.ceph.ibm.com ip=10.18.131.3 ipmi=10.18.139.3
+
+   [maas]
+   ceph-vm-14.tuc.stglabs.ibm.com ip=9.11.120.237
+   ```
+3. **Set Up Secrets**: Encrypt your MAAS credentials in a secrets file (see "Secrets File" section below).
+4. **Run the Playbook**:
+   ```bash
+   ansible-playbook maas-dns-playbook.yml
+   ```
+
+## Variables
+
+### `defaults/main.yml`
+These are overridable defaults:
+- `maas_api_url`: Default MAAS API endpoint (`http://localhost:5240/MAAS/api/2.0/`). Typically overridden by the secrets file.
+- `maas_profile`: Default MAAS profile name (`admin`). Overridden by the secrets file.
+- `allowed_domains`: List of domains to preserve (default: `["maas"]`).
```

Comment:
What does "to preserve" mean?

Reply:
There is a default domain the maas server created called "maas" and I am not sure it can be removed so I added this "allowed_domains" variable to ignore this domain, so preserve means not to delete it

Reply:
I would suggest, instead, having a `default_domains` list in `defaults/main.yml` and have `maas` in it.

Then when deciding which domains to remove, combine the desired domains and the default domains.

Reply:
After checking why there is a default domain called "maas" it turns out that we can remove it so I can completely remove "allowed_domains" and I also don't think we need any default_domains variable in this case

Reply:
I have changed variable name to default_domains, we need to have this variable since the default domain is a dns domain that is used internally by maas, one example is when we deploy a machine it is used by maas for internal dns records so we need to exclude it from our ansible role